client/src/providers/HealthProvider.tsx

```diff
@@ -284,21 +293,69 @@ export const HealthProvider: React.FC<Props> = ({ children, ...props }) => {
       setError('Error Loading Health Contract');
     }
   };
-  const updateProfile = async (hcAddress: string, docAddress: string) => {
+  const updateProfile = async (hcAddress: string, docAddress: string, status: string) => {
     try {
       const web3modal = new Web3Modal();
       const connection = await web3modal.connect();
       const provider = new ethers.providers.Web3Provider(connection);
       const signer = provider.getSigner();
       const contract: HealthBlock = fetchContract(signer);
-      const update = await contract.mapDoctorToProvider(hcAddress, docAddress);
+      let update;
+      if (status == 'confirmed') {
+        update = await contract.mapDoctorToProvider(hcAddress, docAddress);
+      } else {
+        update = await contract.declineDoctorToProviderRequest(hcAddress, docAddress);
+      }
 
       update.wait();
     } catch (err: any) {
       setError(`Error Loading Health Contract ${err}`);
     }
   };
 
+  const fetchAllDoctorToProviderRequests = async (hcAddress: string) => {
+    try {
+      const web3modal = new Web3Modal();
+      const connection = await web3modal.connect();
+      const provider = new ethers.providers.Web3Provider(connection);
+      const signer = provider.getSigner();
+      const contract: HealthBlock = fetchContract(signer);
+      const doctors = await contract.getAllDoctorToProviderRequests(
+        '0xb3cc507e752dcc3da1cef955b58e97ae77160103',
+      );
+      let docArr = [];
+      for (let i = 0; i < doctors.length; i++) {
+        let obj = {
+          name: doctors[i][0],
+          address: doctors[i][1],
+          status: doctors[i][2],
+        };
+        docArr.push(obj);
+      }
+
+      setDocToProviderList(docArr);
+    } catch (err: any) {
+      setError(`Error Loading Health Contract ${err}`);
+    }
+  };
+
+  const raiseDoctorToHCRequest = async (
+    hcAddress: string,
+    docAddress: string,
+    doctorName: string,
+  ) => {
+    try {
+      const web3modal = new Web3Modal();
+      const connection = await web3modal.connect();
+      const provider = new ethers.providers.Web3Provider(connection);
+      const signer = provider.getSigner();
+      const contract: HealthBlock = fetchContract(signer);
+      const update = await contract.raiseDoctorToProviderRequest(hcAddress, docAddress, doctorName);
+    } catch (err: any) {
+      setError(`Error Loading Health Contract ${err}`);
```

Comment:
please dont use this format. these are hooks, return the error to the calling component and let them handle the error.

Reply:
Thank you for the review. I will update it in the next PR.